### DISCUSSION TITLE
Modification keys auto-released on Windows 11

### DISCUSSION THREAD

**Discussion by vladimir-aubrecht** [NicoHood/HID#401] [Q&A]
Hi,

did anybody experience problem with modification keys auto-released on Windows 11?

I am trying to press for example alt+tab, expecting alt+tab dialog will open (and it will). After that I release tab key and I am expecting that dialog will stay on the screen as alt is still hold, but instead alt is released and dialog disappear ...

Problem appears for all modification keys, even shift+letter is often releasing shift before I managed to type letter resulting in lowercased letter. I believe it worked well on Windows 10 (if I remember correctly) and definitely works fine with MacOS, just Windows 11 have this weird problem for some reason ... :-/

Any idea what could be the problem?

```c
NKROKeyboard.press(KEY_LEFT_ALT);
NKROKeyboard.press(KEY_TAB);
NKROKeyboard.send();

NKROKeyboard.release(KEY_TAB);
NKROKeyboard.send();
```

**Comment by NicoHood** [comment]
Your code looks correct. Did you try a short delay between the two calls? Also do you really need the nkro keyboard or maybe the bootkeyboard fits better for you.

**Reply by vladimir-aubrecht** [reply]
I tried with delay of several seconds, but did not help 😕 Also BootKeyboard was behaving in a same way. Thou I discovered, that behaviour is somehow preserved by Windows and code above just starts this behaviour... What I tried:

- I tested commercial keyboard on Windows 11 and experience no auto release problem.
- I connected my Adafruit Feather 32u4 with code above to PC with Windows 11 and after pressing first key (left alt) Windows immediately started to misbehave.
- Disconnecting Feather kept Windows 11 misbehaving until PC restart.

Another thing I noticed is "blinking" address bar in Windows Explorer, like something would be constantly pressing/releasing the key in it even thou nothing is pressed anymore.

Unfortunately don't have much insight into USB protocol, but isn't it possible that Windows are caching some key presses and maybe I am not invalidating something properly?

**Reply by NicoHood** [reply]
Maybe windows has problems with this usb device and how it is constructed. How does the official arduino keyboard library perform?
